tree/ntuple/src/RFieldMeta.cxx

```diff
@@ -366,7 +378,14 @@ void ROOT::RClassField::PrepareStagingArea(const std::vector<const TSchemaRule *
 
    if (stagingAreaSize) {
       R__ASSERT(static_cast<Int_t>(stagingAreaSize) <= fStagingClass->Size()); // we may have removed rules
-      fStagingArea = ROOT::Internal::MakeUninitArray<unsigned char>(stagingAreaSize);
+      // We use std::make_unique instead of MakeUninitArray to zero-initialize the staging area.
```

Comment:
Is the zero-initialize redundant with the constructor calls or do we also need it for the trivial cases?

Reply:
Strictly speaking it may not be necessary for trivial default constructors, but I think it's better to start with a defined value than uninitialized memory. It should not cost since this code is run once when connecting `RClassField` to a page source.